apps/web/src/stores/currentWorkspace.ts

```diff
@@ -42,8 +44,39 @@ export default function useCurrentWorkspace(opts?: SWRConfiguration) {
       mutate(workspace)
       return workspace
     },
-    [mutate],
+    [mutate, data],
+  )
+
+  const { execute: setDefaultProvider } = useServerAction(
```

Comment:
Here if you use `useLatitudeAction` you don't need to deal with form data transformation. Also errors are typed correctly I think.

Reply:
not true, useLatitudeAction expects whatever the underlying server actions expects, and we always use json so it's json.